CMakeLists.txt

```diff
@@ -55,10 +55,17 @@ cmake_dependent_option(ISLE_COMPILE_SHADERS "Compile shaders" ON "SDL_SHADERCROS
 option(CMAKE_POSITION_INDEPENDENT_CODE "Build with -fPIC" ON)
 option(ENABLE_CLANG_TIDY "Enable clang-tidy")
 option(DOWNLOAD_DEPENDENCIES "Download dependencies" ON)
+option(USE_SDL2 "Use SDL2 instead of SDL3 via a shim" OFF)
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}" CACHE PATH "Directory where to put executables and dll")
 set(ISLE_EMSCRIPTEN_HOST "" CACHE STRING "Host URL for Emscripten streaming (e.g., https://test.com)")
 cmake_dependent_option(BUILD_SHARED_LIBS "Build lego1 as a shared library" ON "NOT EMSCRIPTEN" OFF)
 
+if(USE_SDL2)
+  add_compile_definitions(SDL_MAJOR_VERSION=2)
+else()
+  add_compile_definitions(SDL_MAJOR_VERSION=3)
+endif()
```

Comment:
I don't like this defines macro's in SDL's namespace (=`SDL_` prefix).
Something like `ISLE_SDL(2|3)` is safer imho.

Reply:
Are you sure that isn't just because your not including SDL_version.h in the given file? SDL_VERSION_ATLEAST(X, Y, Z) is a standard macro and relies on SDL_MAJOR_VERSION internally.

https://wiki.libsdl.org/SDL2/SDL_VERSION_ATLEAST